test/unit/stop_test.lua

```diff
@@ -0,0 +1,34 @@
+local expirationd = require("expirationd")
+local fiber = require("fiber")
+local t = require("luatest")
+local g = t.group("stop")
+
+local helpers = require("test.helper")
+
+g.before_all(function()
+    helpers.init_spaces(g)
+end)
+
+g.after_each(function()
+    helpers.truncate_spaces(g)
+end)
+
+function g.test_cancel_on_pcall()
+    local function on_full_scan_complete()
+        pcall(fiber.sleep, 1)
+    end
+    local one_hour = 3600
+    local task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {
+        full_scan_delay = one_hour,
+        on_full_scan_complete = on_full_scan_complete
+    })
+    helpers.retrying({}, function()
+        t.assert(task.worker_fiber)
+    end)
+    -- We need to execute in a separate fiber,
+    -- since pcall does not check testcancel and stop may freeze up.
+    local f = fiber.create(task.stop, task)
+    helpers.retrying({}, function()
+        t.assert_equals(f:status(), "dead")
+    end)
```

Comment:
The fixed bug is about freeze on stop, right now in test you just check that fiber some day finished it's work. But we need to check that fiber will finish during  some time. I would suggest to set explicit timeout to `helpers.retrying()`.

Reply:
There is a default timeout in the helpers.retrying(), 5 seconds. Or do you want it to be visible right in the code?

Reply:
Yep, let's make it at least explicit.

Reply:
Made explicit, 5 seconds